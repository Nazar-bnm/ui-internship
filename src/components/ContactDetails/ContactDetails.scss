@import '../../scss/global_styles';

.title{
  @include flex($align: center);
  @include title-font;
  color: $white;
}

.icon.big{
  margin: 4px 3px;
  color: $black;
}

.icon-item{
  border-radius: 50%;
  margin: 3px;
  background-color: $emperor;
}

