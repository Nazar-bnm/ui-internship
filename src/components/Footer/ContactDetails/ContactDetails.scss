@import '@/scss/_global_styles.scss';

.contact-block {
  @include flex($justify: space-between);

  &__title {
    @include flex($align: center);
    @include title-sections;
  }

  &__icons {
    @include flex($justify: center);
    padding-left: 3px;
  }

  &__item {
    border-radius: 50%;
    margin: 6px;
    background-color: $emperor;
    list-style: none;

    &:hover {
      background-color: $white;
    }
  }

  .icon.large {
    margin: 3px 2px;
    color: $black;
  }
}

@include media-for(mobile-landscape) {
  .contact-block {
    &__icons {
      margin-top: 10px;
    }
  }
}
