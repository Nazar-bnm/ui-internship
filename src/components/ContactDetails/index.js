import React from 'react';
import { icons } from '../../config/Footer';
import './ContactDetails.scss';

function ContactDetails() {
  return (
    <div className="container">
      <div className="container col-6">
        <h3 className="title">Connect us</h3>
      </div>
      <div className="container col-6">
        {icons.map(({ name, link }) => (
          <div key={name} className="icon-item">
            <a href={link} target="blank">
              <i className={`${name} icon big`}></i>
            </a>
          </div>
        ))}
      </div>
    </div>
  );
}

export default ContactDetails;
