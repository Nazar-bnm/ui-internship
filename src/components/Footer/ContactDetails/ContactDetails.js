import React from 'react';
import PropTypes from 'prop-types';
import config from '../../../../config';

import './ContactDetails.scss';

const CN = 'contact-block';

const ContactDetails = ({ title }) => {
  const { icons } = config;
  return (
    <div className={`${CN}`}>
      <div className={`${CN}__title`}>
        <h4>{title}</h4>
      </div>
      <ul className={`${CN}__icons`}>
        {icons.map(({ name, link }) => (
          <li key={name} className={`${CN}__item`}>
            <a href={link} target="blank">
              <i className={`${name} icon large`}></i>
            </a>
          </li>
        ))}
      </ul>
    </div>
  );
};

ContactDetails.propTypes = {
  title: PropTypes.string,
};

ContactDetails.defaultProps = {
  title: '',
};

export default ContactDetails;
