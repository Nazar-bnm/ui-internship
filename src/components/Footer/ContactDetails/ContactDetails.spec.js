import React from 'react';
import { shallow } from 'enzyme';
import ContactDetails from './ContactDetails';

describe('<ContactDetails />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<ContactDetails />);
    expect(wrapper).toMatchSnapshot();
  });
});
