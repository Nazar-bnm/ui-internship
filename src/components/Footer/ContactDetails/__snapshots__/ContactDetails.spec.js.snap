// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactDetails /> should match snapshot 1`] = `
<div
  className="contact-block"
>
  <div
    className="contact-block__title"
  >
    <h4 />
  </div>
  <ul
    className="contact-block__icons"
  >
    <li
      className="contact-block__item"
      key="twitter"
    >
      <a
        href="https://twitter.com/griddynamics"
        target="blank"
      >
        <i
          className="twitter icon large"
        />
      </a>
    </li>
    <li
      className="contact-block__item"
      key="mail"
    >
      <a
        href="mailto:khrystynavaskovych@gmail.com"
        target="blank"
      >
        <i
          className="mail icon large"
        />
      </a>
    </li>
    <li
      className="contact-block__item"
      key="facebook"
    >
      <a
        href="https://www.facebook.com/griddynamics"
        target="blank"
      >
        <i
          className="facebook icon large"
        />
      </a>
    </li>
    <li
      className="contact-block__item"
      key="linkedin"
    >
      <a
        href="https://www.linkedin.com/company/grid-dynamics/"
        target="blank"
      >
        <i
          className="linkedin icon large"
        />
      </a>
    </li>
    <li
      className="contact-block__item"
      key="instagram"
    >
      <a
        href="https://www.instagram.com/griddynamics_ua/"
        target="blank"
      >
        <i
          className="instagram icon large"
        />
      </a>
    </li>
    <li
      className="contact-block__item"
      key="google plus g"
    >
      <a
        href="https://plus.google.com/115302417170674279390"
        target="blank"
      >
        <i
          className="google plus g icon large"
        />
      </a>
    </li>
  </ul>
</div>
`;
